src/providers/ChainProvider.tsx

```diff
@@ -37,12 +37,23 @@ export const ChainContext = createContext({} as ContextType);
 export const ChainProvider = ({ children }: StateProps) => {
   const config = useConfig();
   const router = useRouter();
-  const availableChains = useMemo(() => config.chains, [config.chains]);
+  const { isTestnet, isReady } = useTestnet();
+  const availableChains = useMemo(() => {
+    return config.chains.filter((chain: Chain) => (isTestnet ? chain.testnet : !chain.testnet));
+  }, [config.chains, isTestnet]);
   const [fromChain, setFromChain] = useState<Chain>(availableChains[0] as Chain);
```

Comment:
add more break lines to improve the readability of the code
```suggestion
  const { isTestnet, isReady } = useTestnet();
  
  const availableChains = useMemo(() => {
    return config.chains.filter((chain: Chain) => (isTestnet ? chain.testnet : !chain.testnet));
  }, [config.chains, isTestnet]);
  
  const [fromChain, setFromChain] = useState<Chain>(availableChains[0] as Chain);
```